lib/3scale/client.rb

```diff
@@ -52,24 +52,37 @@ class Client
     private_constant :EXTENSIONS_HEADER
 
     def initialize(options)
-      if options[:provider_key].nil? || options[:provider_key] =~ /^\s*$/
-        raise ArgumentError, 'missing :provider_key'
+      @provider_key = options[:provider_key]
+      @service_token = options[:service_token]
+      @use_service_tokens = options[:use_service_tokens]
+
+      if !@use_service_tokens &&
+           [@provider_key, @service_token].all? do |val|
+             val.nil? || val =~ /\A\s*\z/
```

Comment:
I'd say checking for strings with white space is totally unnecessary. Just checking that it is truthy is enough. What would be the use case for this feature?